src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13846,5 +13846,144 @@ int32[] arg2
+                // (5,12): error CS0029: Cannot implicitly convert type '<anonymous delegate>' to 'System.Func<int[], int>'
+                // noParams = withParams;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "withParams").WithArguments("<anonymous delegate>", "System.Func<int[], int>").WithLocation(5, 12),
+                // (6,14): error CS0029: Cannot implicitly convert type 'System.Func<int[], int>' to '<anonymous delegate>'
+                // withParams = noParams;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "noParams").WithArguments("System.Func<int[], int>", "<anonymous delegate>").WithLocation(6, 14));
+        }
+
+        [Fact]
+        public void ParamsArray_LambdaConversion()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                var noParams = MethodNoParams;
+                var withParams = MethodWithParams;
+                noParams = (params int[] xs) => xs.Length;
+                noParams = (int[] xs) => xs.Length;
+                withParams = (params int[] xs) => xs.Length;
+                withParams = (int[] xs) => xs.Length;
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (5,26): error CS9503: Parameter 1 has params modifier in lambda and not in target delegate type.
+                // noParams = (params int[] xs) => xs.Length;
+                Diagnostic(ErrorCode.ERR_ParamsArrayInLambdaOnly, "xs").WithArguments("1").WithLocation(5, 26),
+                // (8,21): error CS9504: Parameter 1 has params modifier in target delegate type and not in lambda.
+                // withParams = (int[] xs) => xs.Length;
+                Diagnostic(ErrorCode.ERR_ParamsArrayInDelegateOnly, "xs").WithArguments("1").WithLocation(8, 21));
+        }
+
+        [Fact]
+        public void ParamsArray_MethodGroupLambdaConversion()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                var lambdaNoParams = (params int[] xs) => xs.Length;
+                var lambdaWithParams = (int[] xs) => xs.Length;
+                lambdaNoParams = MethodWithParams;
+                lambdaWithParams = MethodWithParams;
+                lambdaNoParams = MethodNoParams;
+                lambdaWithParams = MethodNoParams;
+                """;
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ParamsArray_DelegateConversion()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                DelegateNoParams dNoParams;
+                DelegateWithParams dWithParams;
+                dNoParams = (params int[] xs) => xs.Length;
```

Comment:
nit: consider marking lines that get diagnostics with numbered coments (`// 1`, `// 2` and so on). See `LocalScope_11` for example. This helps with readability (matching diagnostics to lines).

Reply:
Thanks, added: 7747e2bb1d6547788b83a588cc0c680bd74597e8